htmlproofer/plugin.py

```diff
@@ -86,47 +88,45 @@ def get_external_url(self, url: str) -> int:
         except requests.exceptions.ConnectionError:
             return -1
 
-    def get_url_status(self, url: str, all_element_ids: Set[str], files: Files,
-                       use_directory_urls: bool) -> Tuple[str, int]:
+    def get_url_status(self, url: str, src_path: str, all_element_ids: Set[str], files: Files,
+                       use_directory_urls: bool) -> int:
         if any(pat.match(url) for pat in LOCAL_PATTERNS):
-            return url, 0
-        clean_url = url.strip('?.')
```

Comment:
I removed this stripping because it breaks relative paths by transforming something like `../../blah.md` to `/../blah.md`. Is there any important utility of this that I'm missing?

Reply:
Well, the commit was three years ago and I can't recall now 😅. Maybe it's not needed now. Can you add a test for your changes ?

Reply:
Added some unit tests to cover the majority of changes in this PR